### ISSUE TITLE
InsertBatch seems unnecessarily fussy about parameters

### ISSUE THREAD

**Issue by ringerc** [pgjdbc/benchmarks#1]
https://github.com/pgjdbc/benchmarks/blob/053464783690bbc35411b8f54ffb71a257f679eb/src/main/java/org/postgresql/benchmark/statement/InsertBatch.java#L70

It seems to only run if multiinsert is 128 or nrows is 1024.

Perhaps something closer to

```
    } else if ((p1nrows % p2multi) != 0) {
      System.err.println("p1nrows must be evenly divisible by p2multi");
      System.exit(-1);
    }
```

was intended?

**Comment by vlsi** [comment]
That is there to save time on measurements.

Otherwise there are too many permutations of the arguments with little additional value.

**Comment by ringerc** [comment]
That makes sense. I wonder if there's a better way to do that, perhaps at the jmh level? Right now, if you want a run like

    java -jar target/benchmarks.jar \
        '^org.postgresql.benchmark.statement.InsertBatch.insertBatch$' \
        -bm Throughput \
        -p p1nrows=100,1000,10000,100000 -p p2multi=1

you can't presently do it.

For now I've patched it in my copy.

I definitely suggest an informative error like "permutation skipped" or "invalid parameter combination"

**Comment by vlsi** [comment]
See https://bugs.openjdk.java.net/browse/CODETOOLS-7901296

**Comment by vlsi** [comment]
I think system property can be added to "run all the permutations".

**Comment by ringerc** [comment]
Makes sense. I think the main thing is to give it a meaningful message and useful comment (issue link, explanation)

**Comment by ringerc** [comment]
It's easy enough to patch if you know what and why
